### PR TITLE
Added keys for the namespace and table metadata

### PR DESCRIPTION
**Pull request by vivekmenezes**
I don't see a strong reason to encode the namespace id in the table metadata key.

### CODE REVIEW

File: keys/constants.go [hunk 1]
```diff
@@ -157,6 +157,10 @@ var (
 	RaftIDGenerator = MakeKey(SystemPrefix, proto.Key("raft-idgen"))
 	// SchemaPrefix specifies key prefixes for schema definitions.
 	SchemaPrefix = MakeKey(SystemPrefix, proto.Key("schema"))
+	// NamespaceMetadataPrefix is the key prefix for all namespace metadata.
+	NamespaceMetadataPrefix = MakeKey(SystemPrefix, proto.Key("ns-"))
```
Comment (tamird) [inline keys/constants.go]: why the dash?

File: keys/keys.go [hunk 2]
```diff
@@ -56,6 +56,16 @@ func NodeStatusKey(nodeID int32) proto.Key {
 	return MakeKey(StatusNodePrefix, encoding.EncodeUvarint(nil, uint64(nodeID)))
 }
 
+// NamespaceMetadataKey returns the key for the namespace.
+func NamespaceMetadataKey(namespace string) proto.Key {
+	return MakeKey(NamespaceMetadataPrefix, proto.Key(namespace))
+}
+
+// TableMetadataKey returns the key for the table in namespaceID.
+func TableMetadataKey(namespaceID uint32, tableName string) proto.Key {
```
Comment (tamird) [inline keys/keys.go]: the other functions in this file start with `Make`

File: keys/keys.go [hunk 3]
```diff
@@ -56,6 +56,16 @@ func NodeStatusKey(nodeID int32) proto.Key {
...
+	return MakeKey(TableMetadataPrefix, encoding.EncodeUvarint(nil, uint64(namespaceID)), proto.Key("-"), proto.Key(tableName))
+}
```
Comment (petermattis) [inline keys/keys.go]: There are no need for separators such as `-` when using the ordered-code encodings. This should be written as:

```
  k := make([]byte, 0, len(TableMetadataPrefix) + len(tableName) + 10)
  k = append(k, TableMetadataPrefix...)
  k = encoding.EncodeUvarint(k, uint64(namespaceID))
  k = append(k, tableName...)
  return k
```

**Comment by petermattis** [comment]
LGTM

File: keys/constants.go (hunk 1 above)
Comment (tamird) [inline keys/constants.go]: did you not see the comment?

File: keys/keys.go [hunk 4]
```diff
@@ -56,6 +56,16 @@ func NodeStatusKey(nodeID int32) proto.Key {
...
 
+// MakeNamespaceMetadataKey returns the key for the namespace.
+func MakeNamespaceMetadataKey(namespace string) proto.Key {
+	return MakeKey(NamespaceMetadataPrefix, proto.Key(namespace))
+}
+
+// MakeTableMetadataKey returns the key for the table in namespaceID.
+func MakeTableMetadataKey(namespaceID uint32, tableName string) proto.Key {
+	return MakeKey(TableMetadataPrefix, encoding.EncodeUvarint(nil, uint64(namespaceID)), proto.Key(tableName))
+}
```
Comment (petermattis) [inline keys/keys.go]: Still recommend restructuring this as I suggested.

**Pull request merged**